interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,764 @@
+    mechanism_text.append(build_reactions_text(all_reactions))
+
+    if transport_path is None and transport_exists:
+        mechanism_text.append(build_transport_text(all_species, separate_file=False))
+    elif transport_path is not None and transport_exists:
+        transport_text = [
+            header_text,
+            build_transport_text(all_species, separate_file=True),
+        ]
+        transport_path.write_text("\n".join(transport_text))
+
+    mechanism_path.write_text("\n".join(mechanism_text))
+
+    return output_files
+
+
+def main():
+    """
+    Parse command line arguments and pass them to `convert`
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Convert Cantera YAML input files to Chemkin-format mechanisms",
+    )
+
+    parser.add_argument("input", help="The input YAML filename. Required.")
+    parser.add_argument(
+        "--phase-name",
+        help=(
+            "Name of the phase to load from the input. If unspecified, the first "
+            "phase listed in the input file will be loaded."
+        ),
+        metavar="PHASE",
+        default="",
+    )
+
+    parser.add_argument(
+        "--mechanism",
+        metavar="MECH",
+        help=(
+            "The path to the output mechanism file. Optional. If not provided, the "
+            "name of the input file will be used, with the extension replaced by '.ck'."
+            "If the path specified here is an existing directory, the output file(s) "
+            "will be placed in that directory."
+        ),
+    )
+    parser.add_argument(
+        "--thermo",
+        metavar="THERM",
+        help=(
+            "The path to the output thermodynamics database file. Optional. If not "
+            "provided, the thermodynamic data will be included in the mechanism file."
+        ),
+    )
+    parser.add_argument(
+        "--transport",
+        metavar="TRAN",
+        help=(
+            "The path to the output transport database file. Optional. If not "
+            "provided, transport data will be included in the mechanism file."
+        ),
+    )
+    parser.add_argument(
+        "--sort-elements",
+        choices=[None, "alphabetical", "molar-mass"],
+        default=None,
+        help=(
+            "Sort elements in source order (None), alphabetically, or from lowest to "
+            "highest atomic mass."
+        ),
+    )
+    parser.add_argument(
+        "--sort-species",
+        choices=[None, "alphabetical", "molar-mass"],
+        default=None,
+        help=(
+            "Sort species in source order (None), alphabetically, or from lowest to "
+            "highest molecular weight."
+        ),
+    )
+    parser.add_argument(
+        "--overwrite",
+        action="store_true",
+        default=False,
+        help="If set, overwrite existing output files.",
+    )
+    parser.add_argument(
+        "--validate",
+        action="store_true",
+        default=True,
```

Comment:
```suggestion
        "--no-validate",
        action="store_true",
        default=False,
```
Otherwise validate would be run in all cases?

Reply:
Ah, with the way this was before, you could do `--validate=False` to turn off validation, but I think it probably makes sense to have `--validate` and `--no-validate`

Reply:
if validation is the default, then `--validate` isn't needed?

Reply:
There's a nice action (present in 3.9+) that automatically handles this kind of thing. I'll add that. I think this kind of behavior is typical for optional Boolean arguments.